src/main/java/com/palantir/gradle/revapi/RevapiReportTask.java

```diff
@@ -17,23 +17,46 @@
 package com.palantir.gradle.revapi;
 
 import com.google.common.collect.ImmutableMap;
-import com.google.common.io.Resources;
 import com.palantir.gradle.revapi.config.Justification;
-import java.io.File;
-import java.io.IOException;
+import freemarker.cache.ClassTemplateLoader;
+import freemarker.template.Configuration;
+import freemarker.template.DefaultObjectWrapperBuilder;
+import freemarker.template.Template;
+import java.io.StringWriter;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Map;
-import java.util.Optional;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
 import org.gradle.api.tasks.TaskAction;
 
-public class RevapiReportTask extends RevapiJavaTask {
+public class RevapiReportTask extends DefaultTask {
+    private final RegularFileProperty resultsFile = getProject().getObjects().fileProperty();
+    private final RegularFileProperty junitOutputFile = getProject().getObjects().fileProperty();
+
+    public RevapiReportTask() {
+        this.getOutputs().upToDateWhen(_task -> false);
+    }
+
+    @InputFile
+    public final RegularFileProperty getResultsFile() {
+        return resultsFile;
+    }
+
+    @OutputFile
```

Comment:
i'd just not make this task cacheable at all, so no need for input/output annotations